apps/opik-backend/src/main/java/com/comet/opik/domain/llmproviders/OpenAi.java

```diff
@@ -0,0 +1,108 @@
+package com.comet.opik.domain.llmproviders;
+
+import com.comet.opik.infrastructure.LlmProviderClientConfig;
+import dev.ai4j.openai4j.OpenAiClient;
+import dev.ai4j.openai4j.OpenAiHttpException;
+import dev.ai4j.openai4j.chat.ChatCompletionRequest;
+import dev.ai4j.openai4j.chat.ChatCompletionResponse;
+import io.dropwizard.jersey.errors.ErrorMessage;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.ClientErrorException;
+import jakarta.ws.rs.InternalServerErrorException;
+import jakarta.ws.rs.ServerErrorException;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+
+@Slf4j
+public class OpenAi implements LlmProviderService {
+    private static final String UNEXPECTED_ERROR_CALLING_LLM_PROVIDER = "Unexpected error calling LLM provider";
+
+    private final LlmProviderClientConfig llmProviderClientConfig;
+    private final OpenAiClient openAiClient;
+
+    @Inject
+    public OpenAi(LlmProviderClientConfig llmProviderClientConfig, String apiKey) {
+        this.llmProviderClientConfig = llmProviderClientConfig;
+        this.openAiClient = newOpenAiClient(apiKey);
+    }
+
+    @Override
+    public ChatCompletionResponse generate(@NonNull ChatCompletionRequest request, @NonNull String workspaceId) {
```

Comment:
I'm assuming (but not confident) that most of this code will be similar for most other providers.  We can wait until we have another provider to support, but when we implement this for another provider pay attention if there are many similarities and a common parent can be extracted

Reply:
I'm starting to work on the next integration at the moment in the same branch. We can see then.
At first glance before starting to work on this, the interfaces are quite different and this couldn't be generalized further.